mmv1/third_party/terraform/services/compute/resource_compute_instance_group_membership_test.go.erb

```diff
@@ -0,0 +1,229 @@
+    }
+  `, context) + testAccComputeInstanceGroupMembership_noInstanceGroupMembership(context)
+}
+
+func testAccComputeInstanceGroupMembership_instanceGroupMembershipAdditional(context map[string]interface{}) string {
+  return acctest.Nprintf(`
+    resource "google_compute_instance_group_membership" "add1" {
+      instance_group = google_compute_instance_group.default.name
+      instance = google_compute_instance.add1.self_link
+    }
+
+    resource "google_compute_instance_group_membership" "add2" {
+      instance_group = google_compute_instance_group.default.name
+      instance = google_compute_instance.add2.self_link
+    }
+  `, context) + testAccComputeInstanceGroupMembership_instanceGroupMembershipBasic(context)
+}
+
+func testAccComputeInstanceGroupMembership_noInstanceGroupMembership(context map[string]interface{}) string {
+  return acctest.Nprintf(`
+    resource "google_compute_network" "default-network" {
+      name = "default-%{random_suffix}"
+    }
+
+    resource "google_compute_instance" "default" {
+      name         = "default-instance-%{random_suffix}"
+      machine_type = "e2-medium"
+
+      boot_disk {
+        initialize_params {
+          image = "debian-cloud/debian-11"
+        }
+      }
+
+      network_interface {
+        network = google_compute_network.default-network.name
+      }
+    }
+
+    resource "google_compute_instance" "add1" {
+      name         = "add1-instance-%{random_suffix}"
+      machine_type = "e2-medium"
+
+      boot_disk {
+        initialize_params {
+          image = "debian-cloud/debian-11"
+        }
+      }
+
+      network_interface {
+        network = google_compute_network.default-network.name
+      }
+    }
+
+    resource "google_compute_instance" "add2" {
+      name         = "add2-instance-%{random_suffix}"
+      machine_type = "e2-medium"
+
+      boot_disk {
+        initialize_params {
+          image = "debian-cloud/debian-11"
+        }
+      }
+
+      network_interface {
+        network = google_compute_network.default-network.name
+      }
+    }
+
+    resource "google_compute_instance_group" "default" {
+      name      = "instance-group-%{random_suffix}"
+    }
+  `, context)
+}
+
+func testAccCheckComputeInstanceGroupMembershipDestroyed(t *testing.T, instanceGroupId string, instances ...string) resource.TestCheckFunc {
+  return func(s *terraform.State) error {
+    foundInstances, err := testAccComputeRegionInstanceGroupMembershipListMembership(t, instanceGroupId)
+    if err != nil {
+      return fmt.Errorf("unable to confirm instance group members with instances %+v was destroyed: %v", instances, err)
+    }
+    for _, p := range instances {
+      if _, ok := foundInstances[p]; ok {
+        return fmt.Errorf("instance group with instance %s still exists", p)
+      }
+    }
+    return nil
+  }
+}
+
+func testAccComputeRegionInstanceGroupMembershipListMembership(t *testing.T, instanceGroupId string) (map[string]struct{}, error) {
```

Comment:
Nit: Is "Region" relevant here (and for `testAccComputeRegionInstanceGroupMembershipGetInstanceName`)? Mainly mentioning to prevent potential name conflicts.